Testing/models

juanmbellini opened the pull request:
# Summary

This PR adds testing over the models, according to the trello card: [https://trello.com/c/HVp5JpUa](https://trello.com/c/HVp5JpUa). Is the continuation of #12 .

# Trello Card

https://trello.com/c/HVp5JpUa

juanmbellini committed:
Extract mock values generation to another class

juanmbellini committed:
Add testing for the Book model

juanmbellini committed:
Add testing for the User model

darkBasic666 commented on src/test/java/wolox/training/models/BookTest.java:
```diff
@@ -0,0 +1,196 @@
+    @DisplayName("Create Book - Valid arguments")
+    void testCreateBookWithValidArguments() {
+        Assertions.assertDoesNotThrow(
+            () -> new Book(
```
maybe to no repeat always constructor with all params, you can use a hash map.

juanmbellini replied:
Done. Check the `ModelsTestHelper` class.

juanmbellini committed:
Add a ModelsTestHelper class

- Include methods to create a Book from a Map with values to be used, together with a method to create the said Map with valid values

juanmbellini committed:
Improve the BookTest class

- Use the ModelsTestHelper class to avoid too much text when creating a Book

juanmbellini committed:
Improve the UserTest class and add User creation helpers to ModelsTestHelper

- Make use of the Map approach to avoid too much text

juanmbellini committed:
Add a private constructor for the ModelsTestHelper class

- This avoids instantiation
